### PR TITLE
Added UUID entity.

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Signed-off-by: Tomas Kraus <tomas.kraus@oracle.com>

---
name: Pull Request
about: Create a pull request for a Platform TCK change
title: JPA UUID Test
labels:
assignees: Tomas-Kraus

---

**Fixes Issue**
https://github.com/eclipse-ee4j/jpa-api/pull/319

**Related Issue(s)**
https://github.com/eclipse-ee4j/jpa-api/issues/151
https://github.com/eclipse-ee4j/jpa-api/issues/152

**Describe the change**
Simple test to verify EM persist of an entity class with UUID.

**Additional context**
N/A

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @scottmarlow @lukasj

### CODE REVIEW

**Comment by Tomas-Kraus** [comment]
Keeping this as draft. This PR requires UUID JPA spec change to be merged and JPA Spec Assertion IDs to be updated.

**Comment by gavinking** [comment]
Nice.

**Comment by jgrassel** [comment]
(heard that my code review comment isn't visible):
Shouldn't there be a call to EntityManager.clear() after the flush() to force the provider-impl to fetch from the database? Otherwise, the provider-impl will just short-circuit to fetching what's already in the persistence context and will not have to make a fetch request to the database.

**Comment by Tomas-Kraus** [comment]
@jgrassel I did not see your comment in the code. But adding clear is not a problem. I can do it with Spec Assertion IDs update.

**Comment by Tomas-Kraus** [comment]
This PR contains JPA 3.1 spec assertions IDs update.

**Comment by Tomas-Kraus** [comment]
JPA API 3.1.0-RC2 is in central. Test must be verified using this latest API.

**Review by scottmarlow — APPROVED** [review]

File: src/com/sun/ts/tests/jpa/core/types/field/Client.java [hunk 1]
```diff
@@ -1018,6 +1019,40 @@ public void fieldTypeTest17() throws Fault {
...
+   *
+   * @test_Strategy: Test EM create on entity class with UUID.
+   */
+  public void testCreateUUIDType() throws Fault {
+    UUID id = UUID.randomUUID();
+    UUIDType uuidType = new UUIDType(id, "Create UUID Type");
+    try {
+      getEntityTransaction().begin();
+      getEntityManager().persist(uuidType);
+      getEntityManager().flush();
```
Comment (scottmarlow) [inline src/com/sun/ts/tests/jpa/core/types/field/Client.java:1035]: As per[ comment about adding call to `EntityManager.clear()`](https://github.com/eclipse-ee4j/jakartaee-tck/pull/793#issuecomment-1011295379), should a call to `getEntityManager().clear()` be added here?

**Review by scottmarlow — CHANGES_REQUESTED** [review]
testCreateUUIDType() seems to be missing the call to EntityManager.clear() which from previous comments, I expected to see added.

File: src/com/sun/ts/tests/jpa/core/types/field/Client.java (hunk 1 above)
Comment (Tomas-Kraus) [reply src/com/sun/ts/tests/jpa/core/types/field/Client.java:1035]: Done.

**Comment by Tomas-Kraus** [comment]
Yes, I forgot it. Should be OK now.
@alwin-joseph Alwin, please can you test this with JPA API 3.1.0-RC2?

**Review by scottmarlow — APPROVED** [review]

**Pull request merged**